### PR TITLE
Fixing Model animation bugs

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/17998/Changing-a-model-url-on-an-animated-item-stops-the-animation-and-it-cannot-be-recovered
https://highfidelity.manuscript.com/f/cases/14989/Changing-from-a-valid-model-URL-to-a-blank-one-does-not-display-green-bounding-box
https://highfidelity.manuscript.com/f/cases/18274/Editing-animated-models-appears-to-cause-a-heap-corruption-and-crash-the-client

Test plan:
- For [17998](https://highfidelity.manuscript.com/f/cases/17998/Changing-a-model-url-on-an-animated-item-stops-the-animation-and-it-cannot-be-recovered), follow the instructions in the ticket.  After changing the URL, the new model should still animate.
- For [14989](https://highfidelity.manuscript.com/f/cases/14989/Changing-from-a-valid-model-URL-to-a-blank-one-does-not-display-green-bounding-box), follow the instructions in the ticket.  Models with valid URLs should render.  Changing to an invalid URL should render a green box (after a few seconds because it takes time for the download to fail).  Changing to an empty URL should render a green box instantly.  Go between all 3 states in different orders and the box should render or not render correctly each time.
- For [18274](https://highfidelity.manuscript.com/f/cases/18274/Editing-animated-models-appears-to-cause-a-heap-corruption-and-crash-the-client), import the full content set from the ticket.  Get multiple people to stand in it.  You shouldn't crash.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12818/HighFidelity-Beta-PR13969-9a2bdda.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12818/HighFidelity-Beta-PR13969-9a2bdda.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12818/HighFidelity-Beta-Interface-PR13969-9a2bdda.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12818/HighFidelity-Beta-PR13969-9a2bdda.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12818/HighFidelity-Beta-Interface-PR13969-9a2bdda.exe

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -1311,6 +1311,8 @@ void ModelEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& sce
             emit DependencyManager::get<scriptable::ModelProviderFactory>()->
                 modelRemovedFromScene(entity->getEntityItemID(), NestableType::Entity, _model);
         }
+        setKey(false);
+        _didLastVisualGeometryRequestSucceed = false;
```
Comment (SamGondelman) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: This makes models with no URL render as green boxes.

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 2]
```diff
@@ -1336,6 +1338,7 @@ void ModelEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& sce
     if (_parsedModelURL != model->getURL()) {
         withWriteLock([&] {
             _texturesLoaded = false;
+            _jointMappingCompleted = false;
```
Comment (SamGondelman) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: This makes animations work when you switch model URLs

File: libraries/entities/src/ModelEntityItem.cpp [hunk 3]
```diff
@@ -581,6 +592,18 @@ void ModelEntityItem::setAnimationCurrentFrame(float value) {
...
+        _animationProperties.setAllowTranslation(value);
+    });
+}
+
+bool ModelEntityItem::getAnimationAllowTranslation() const {
+    return resultWithReadLock<bool>([&] {
+        return _animationProperties.getAllowTranslation();
+    });
+}
+
```
Comment (SamGondelman) [inline libraries/entities/src/ModelEntityItem.cpp]: The changes in this file are to make the animation properties threadsafe.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12931/HighFidelity-Beta-PR13969-42ded53.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12931/HighFidelity-Beta-PR13969-42ded53.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12931/HighFidelity-Beta-Interface-PR13969-42ded53.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12931/HighFidelity-Beta-PR13969-42ded53.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12931/HighFidelity-Beta-Interface-PR13969-42ded53.exe

**Comment by qaAdamSmith82** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13045/HighFidelity-Beta-PR13969-42ded53.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13045/HighFidelity-Beta-PR13969-42ded53.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13045/HighFidelity-Beta-Interface-PR13969-42ded53.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13045/HighFidelity-Beta-PR13969-42ded53.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13045/HighFidelity-Beta-Interface-PR13969-42ded53.exe

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by qaAdamSmith82** [comment]
Pass @conklin94122 

All bugs resolved

**Pull request merged**